Jenkinsfile

```diff
@@ -20,6 +20,117 @@
 /* groovylint-disable-next-line CompileStatic */
 job_status_internal = [:]
 
+/**
+ * unitTestPostEx step method
+ *
+ * @param config Map of parameters passed
+ *
+ * config['always_script']       Script to run after any test.
+ *                               Default 'ci/unit/test_post_always.sh'.
+ *
+ * config['ignore_failure']      Ignore test failures.  Default false.
+ *
+ * config['referenceJobName']    Reference job name.
+ *                               Defaults to 'daos-stack/daos/master'
+ *
+ * config['testResults']         Junit test result files.
+ *                               Default 'test_results/*.xml'
+ *
+ * config['valgrind_pattern']    Pattern for Valgind files.
+ *                               Default: '*.memcheck.xml'
+ *
+ * config['valgrind_stash']      Name to stash valgrind artifacts
+ *                               Required if more than one stage is
+ *                               creating valgrind reports.
+ * config['tgz_file']            Name of tgz file for valgrind results
+ */
+def unitTestPostEx(Map config = [:], List artifacts) {
+  String always_script = config.get('always_script', 'ci/unit/test_post_always.sh')
+  boolean ignore_failure = config.get('ignore_failure', false)
+  String referenceJobName = config.get('referenceJobName', 'daos-stack/daos/master')
+  String testResults = config.get('testResults', 'test_results/*.xml')
+  String valgrind_pattern = config.get('valgrind_pattern', '*.memcheck.xml')
+  String valgrind_stash = config.get('valgrind_stash', null)
+  String tgz_file = config.get('tgz_file', 'unit_test_memcheck_logs.tar.gz')
+
+  sh label: 'Job Cleanup',
+     script: always_script
+
+  Map stage_info = parseStageInfo(config)
+
+  if (testResults != 'None' ) {
+    double health_scale = 1.0
+    if (ignore_failure) {
+      health_scale = 0.0
+    }
+
+    def cb_result = currentBuild.result
+    junit testResults: testResults,
+          healthScaleFactor: health_scale
+
+    if (cb_result != currentBuild.result) {
+      println "The junit plugin changed result to ${currentBuild.result}."
```

Comment:
We have a hidden wrapper for junit that does the cb_result check and logs the result, so the cb_result check here is not needed.

Reply:
This is copied from pipeline-lib and I'd like to limit changes until subsequent patches.